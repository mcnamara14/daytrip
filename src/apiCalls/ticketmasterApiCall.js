import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';

export const ticketmasterApiCallRecentEvents = async (city, state, timeNow, timeIn2Days) => {
  const url = `https://app.ticketmaster.com/discovery/v2/events.json?startDateTime=${timeNow}&endDateTime=${timeIn2Days}&sort=date,name,asc&city=${city}&stateCode=${state}&apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;

    return events
  } catch (error) {
      alert(error)
  }
}





