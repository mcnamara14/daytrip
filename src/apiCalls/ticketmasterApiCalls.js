import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';
import { 
  cleanRecentEventsSearch, 
  cleanRecentEvents 
} from '../dataCleaners/index';

export const fetchRecentEvents = async (city, state, timeNow) => {
  const prefix = 'https://app.ticketmaster.com/discovery/v2/events.json?';
  const start = `startDateTime=${timeNow}`;
  const sort = 'sort=date,name,asc';
  const userCity = `city=${city}`;
  const userState = `stateCode=${state}`;
  const apiKey = `apikey=${ticketmasterApiKey}`;

  const url = `${prefix}&${start}&${sort}&${userCity}&${userState}&${apiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;
    console.log(events)

    return events;
  } catch (error) {
    alert(error);
  }
};

export const fetchRecentEventsOnSearch = async (
  userCity, 
  userState, 
  timeNow, 
  keyword) => {
  const prefix = 'https://app.ticketmaster.com/discovery/v2/events.json?';
  const start = `startDateTime=${timeNow}`;
  const sort = 'sort=date,name,asc';
  const city = `city=${userCity}`;
  const state = `stateCode=${userState}`;
  const input = `keyword=${keyword}`;
  const key = `apikey=${ticketmasterApiKey}`;

  const url = `${prefix}${start}&${sort}&${city}&${state}&${input}&${key}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;
  
    return cleanRecentEventsSearch(events);
  } catch (error) {
    console.log(error)
  }

};

export const fetchSelectedEvent = async (id) => {
  const prefix = `https://app.ticketmaster.com/discovery/v2/events/${id}.json?`;
  const url = `${prefix}apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const event = await response.json();
    
    return cleanRecentEvents([event]);
  } catch (error) {
    console.log('error');
  }
};










