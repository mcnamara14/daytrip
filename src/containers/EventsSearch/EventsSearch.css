* {
  font-family: "Work Sans";
}

.eventsContainer {
    margin-top: -115px;
}

.eventsSearchContainer {
  width: calc(40% - 100px);
  height: 300px;
  background: #fff;
  text-align: left;
  padding: 20px 100px 20px 80px;
}

.eventsSearchContainer button {
  height: 50px;
  width: 41%;
  border-radius: 3px;
  color: #fff;
  background: #292829;
  font-size: 0.9rem;
  border: 1px solid #ccc;
  margin-top: 20px;
}

.eventsSearchContainer h5 {
  color: #7c7c7c;
  font-weight: 300;
  font-size: 1.5rem;
  margin-bottom: 20px;
}

.eventsSearchContainer input {
  width: 100%;
  height: 45px;
  border-radius: 3px;
  border: 1px solid #ccc;
  padding: 0 0 0 20px;
  font-size: 1rem;
}

.eventsSearchLocationDate {
  display: flex;
  margin-top: 30px;
  justify-content: space-between;
}

.eventsSearchLocationDate input, .eventsSearchLocationDate div {
  width: 46%;
}

.eventsSearchLocationDate .react-datepicker-wrapper, .eventsSearchLocationDate .react-datepicker__input-container input, .eventsSearchLocationDate .react-datepicker__input-container {
  width: 96% !important;
  color: #7c7c7c;
}

.react-datepicker {
  width: 80% !important;
}

.react-datepicker__day-name {
  width: 22% !important;
  margin: 0.4rem;
}

.react-datepicker__header, .react-datepicker__month-container {
  width: 100% !important;
}

.Select-input {
  height: 45px !important;
}

.Select-input > input {
  padding: 0 !important;
}

.Select.has-value.is-clearable.Select--single > .Select-control .Select-value {
  display: flex;
  align-items: center;
}

.Select-placeholder {
  display: flex;
  align-items: center;
}