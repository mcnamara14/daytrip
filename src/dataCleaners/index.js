import { cleanRecentEvents } from './recentEventsCleaner';

export {
  cleanRecentEvents
}