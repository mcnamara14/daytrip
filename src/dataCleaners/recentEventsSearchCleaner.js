export const cleanRecentEventsSearch = (events) => {
  const eventsSearchResults = events.map((event, index) => {
    const {
      id,
      name,
      _embedded
    } = event;

    const venue = _embedded.venues[0].name;

    return {
      value: index,
      label: name + venue,
      id
    };
  });

  return {options: eventsSearchResults};
};