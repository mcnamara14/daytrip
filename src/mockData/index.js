import { mockDirtyRecentEvents } from './mockRecentEvents';
import { mockCleanRestaurantAndBarData } from './mockRestaurantAndBarData';

export {
  mockDirtyRecentEvents,
  mockCleanRestaurantAndBarData
}