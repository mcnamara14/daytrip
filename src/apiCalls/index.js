import { ticketmasterApiCall } from './ticketmasterApiCall'; 
import { yelpApiCall } from './yelpApiCall'; 

export {
  ticketmasterApiCall,
  yelpApiCall
};