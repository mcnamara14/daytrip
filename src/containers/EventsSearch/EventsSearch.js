import React, { Component } from 'react';
import { connect } from 'react-redux';
import './EventsSearch.css';
import LocationAutocomplete from 'location-autocomplete';
import { googleApiKey } from '../../apiCalls/apiKeys/googleApiKey';
import Calendar from 'rc-calendar';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import 'react-datepicker/dist/react-datepicker.css';
import { Async } from 'react-select';
import { cleanRecentEventsSearch, cleanRecentEvents } from '../../dataCleaners/index';
import { ticketmasterApiCallRecentEventsSearch, ticketmasterFetchSelectedEvent } from '../../apiCalls/index';
import 'react-select/dist/react-select.css';
import { storeSelectedEvent } from '../../actions/index';
import { select } from 'redux-saga/effects';

class EventsSearch extends Component {
  constructor() {
    super();

    this.state = {
      startDate: moment(),
      location: '',
      city: '',
      state:'',
      locationError: false,
      selectedOption: 'Enter a location'
    };
  }

  handleChange = (date) => {
    this.setState({
      startDate: date
    });
  }

  componentWillMount() {
    if(this.props.user) {
      const selectedOption = this.props.user.city + ',' + this.props.user.state;

      this.setState({
        city: this.props.user.city,
        state: this.props.user.state,
        selectedOption
      })
    }
  }

  handleTicketMasterFetch = async (input) => {
    if (this.state.city && this.state.state) {
      const city = this.state.city;
      const state = this.state.state;
      const date = this.state.startDate;
      const timeNow = date.format();
      const events = await ticketmasterApiCallRecentEventsSearch(city, state, timeNow, input);
      console.log(events)
      return events;
    } else {
      this.handleMissingLocationError();
    }
  }

  onSelect = (selectedOption) => {
    this.setState({ selectedOption });
		
    if (selectedOption) {
      console.log(`Selected: ${selectedOption.id}`);
    }
  }

  handleMissingLocationError = () => {
    this.setState({locationError: true});
    setTimeout(() => {
      this.setState({
        locationError: false
      });
    }, 2000);
  }
  
  handleStoreEvent = async () => {
    const eventId = this.state.selectedOption.id;
    const selectedEvent = await ticketmasterFetchSelectedEvent(eventId);
    const event = selectedEvent[0]

    this.props.storeSelectedEvent(event);
  }

  onDropdownSelect = (component) => {
    const place = component.autocomplete.getPlace();
    const city = place.vicinity;
    const state = place.address_components[2].short_name;
    
    this.setState({
      city,
      state
    });
  }

  render() {
    const { selectedOption } = this.state;
 
    return (
      <div className="eventsSearchContainer">
        <h5>Search Events</h5>
        <Async
          name="searchEventsInput"
          loadOptions={this.handleTicketMasterFetch}
          value={selectedOption}
          onChange={this.onSelect}
          placeholder="Search events"
        />
        <div className="eventsSearchLocationDate">
          <LocationAutocomplete
            name="location"
            placeholder={this.state.selectedOption ? this.state.selectedOption : 'Enter a location'}
            targetArea="City, State"
            locationType="(cities)"
            onChange={this.onChangeHandler}
            googleAPIKey={googleApiKey} 
            onDropdownSelect={this.onDropdownSelect}
          />
          <DatePicker
            selected={this.state.startDate}
            onChange={this.handleChange}
          />
        </div>
        <button onClick={this.handleStoreEvent} >Select Event</button>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  storeSelectedEvent: (event) => dispatch(storeSelectedEvent(event))
})

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps, mapDispatchToProps)(EventsSearch);