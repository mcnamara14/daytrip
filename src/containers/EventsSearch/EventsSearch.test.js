import React from 'react';
import ReactDOM from 'react-dom';
import { EventsSearch } from './EventsSearch';
import { shallow } from 'enzyme';
import moment from 'moment';
// jest.mock('moment', () => ({format: () => '2018–01–30T12:34:56+00:00'}));
import { mockDirtyRecentEvents, mockUser } from '../../mockData';
import { cleanRecentEventsSearch } from '../../dataCleaners/recentEventsSearchCleaner';
jest.mock('../../dataCleaners/recentEventsSearchCleaner');
import { fetchRecentEventsOnSearch } from '../../apiCalls';
jest.mock('../../apiCalls');


describe('EventsSearch', () => {
  it('should match the snapshot', () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        _embedded:
        mockDirtyRecentEvents
      })
    }));
    const mockEvents = mockDirtyRecentEvents.events;
    const wrapper = shallow(<EventsSearch events={mockEvents} user={mockUser} />);
    expect(wrapper).toMatchSnapshot();
  });


  describe('handleChange', () => {
    it('should set data in state to selected date', () => {
      const wrapper = shallow(<EventsSearch user={mockUser} />);
      
      wrapper.instance().handleChange('062018');

      const expected = '062018';
      expect(wrapper.state('startDate')).toEqual(expected)
    })
  });

  // Cant figure out how to mock moment.js format method

  describe('fetchEvents', () => {
    // it('should call fetchRecentEventsOnSearch with the correct arguments if a location is entered', async () => {
    //   const wrapper = shallow(<EventsSearch user={mockUser} />);
    //   wrapper.setState({
    //     city: 'Boulder', 
    //     state: 'CO',
    //     startDate: '062018'
    //   });

    //   wrapper.instance().fetchEvents('rockies');
    //   const result = await fetchRecentEventsOnSearch();

    //   expect(fetchRecentEventsOnSearch).toHaveBeenCalledWith(0)
    // })

    it('should call handleMissingLocationError when there is no location', () => {
      const wrapper = shallow(<EventsSearch user={mockUser} />);

      wrapper.setState({
        city: '', 
        state: '',
        startDate: '062018'
      });

      wrapper.instance().handleMissingLocationError = jest.fn();

      const result = wrapper.instance().handleMissingLocationError

      wrapper.instance().fetchEvents('rockies');

      expect(result).toHaveBeenCalled();
    })
  });


});
