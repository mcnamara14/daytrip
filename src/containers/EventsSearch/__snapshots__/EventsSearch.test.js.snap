// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsSearch should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventsSearch
    events={
        Array [
            Object {
              "_embedded": Object {
                "venues": Array [
                  Object {
                    "location": Object {
                      "latitude": "39.735001",
                      "longitude": "-105.020401",
                    },
                    "name": "Boulder Theater",
                  },
                ],
              },
              "classifications": Array [
                Object {
                  "test": "...",
                },
              ],
              "dates": Object {
                "start": Object {
                  "localDate": "2018-05-30",
                  "localTime": "08:00:00",
                },
              },
              "id": "Zrt9dk32l",
              "images": Array [
                Object {
                  "url": "https://s1.ticketm.allswifty.jpg",
                },
              ],
              "name": "T Swift",
              "priceRanges": Array [
                Object {
                  "currency": "USD",
                  "max": 12,
                  "min": 300,
                },
              ],
              "sales": Object {
                "public": Object {
                  "test": "...",
                },
              },
              "type": "event",
              "url": "http://www.ticketsnow.com/PID=2107079",
            },
          ]
    }
    user={
        Object {
            "city": "San Diego",
            "email": "mcnamara14@gmail.com",
            "state": "CA",
            "userId": "tvcmzDN3SoZ6vU07a1w9vY3S8zt1",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Search Events
</h5>,
        <Async
          autoload={false}
          cache={Object {}}
          ignoreAccents={true}
          ignoreCase={true}
          loadOptions={[Function]}
          loadingPlaceholder="Loading..."
          name="searchEventsInput"
          onChange={[Function]}
          options={Array []}
          placeholder="Search events"
          searchPromptText="Type to search"
          value="San Diego,CA"
>
          [Function]
</Async>,
        <div
          className="eventsSearchLocationDate"
>
          
          <LocationAutocomplete
                    googleAPIKey="AIzaSyCsk3J8WFJ5LEMPbhNooarHGiQ-vrrbZGE"
                    locationType="(cities)"
                    name="location"
                    onChange={undefined}
                    onDropdownSelect={[Function]}
                    placeholder="San Diego,CA"
                    targetArea="City, State"
          />
          <DatePicker
                    allowSameDay={false}
                    dateFormat="L"
                    dateFormatCalendar="MMMM YYYY"
                    disabled={false}
                    disabledKeyboardNavigation={false}
                    dropdownMode="scroll"
                    monthsShown={1}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClickOutside={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMonthChange={[Function]}
                    onSelect={[Function]}
                    onYearChange={[Function]}
                    preventOpenOnFocus={false}
                    selected={"2018-06-06T04:03:09.129Z"}
                    shouldCloseOnSelect={true}
                    showTimeSelect={false}
                    timeCaption="Time"
                    timeIntervals={30}
                    withPortal={false}
          />
</div>,
        <button
          onClick={[Function]}
>
          Select Event
</button>,
      ],
      "className": "eventsSearchContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search Events",
        },
        "ref": null,
        "rendered": "Search Events",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoload": false,
          "cache": Object {},
          "children": [Function],
          "ignoreAccents": true,
          "ignoreCase": true,
          "loadOptions": [Function],
          "loadingPlaceholder": "Loading...",
          "name": "searchEventsInput",
          "onChange": [Function],
          "options": Array [],
          "placeholder": "Search events",
          "searchPromptText": "Type to search",
          "value": "San Diego,CA",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <LocationAutocomplete
              googleAPIKey="AIzaSyCsk3J8WFJ5LEMPbhNooarHGiQ-vrrbZGE"
              locationType="(cities)"
              name="location"
              onChange={undefined}
              onDropdownSelect={[Function]}
              placeholder="San Diego,CA"
              targetArea="City, State"
/>,
            <DatePicker
              allowSameDay={false}
              dateFormat="L"
              dateFormatCalendar="MMMM YYYY"
              disabled={false}
              disabledKeyboardNavigation={false}
              dropdownMode="scroll"
              monthsShown={1}
              onBlur={[Function]}
              onChange={[Function]}
              onClickOutside={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMonthChange={[Function]}
              onSelect={[Function]}
              onYearChange={[Function]}
              preventOpenOnFocus={false}
              selected={"2018-06-06T04:03:09.129Z"}
              shouldCloseOnSelect={true}
              showTimeSelect={false}
              timeCaption="Time"
              timeIntervals={30}
              withPortal={false}
/>,
          ],
          "className": "eventsSearchLocationDate",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "googleAPIKey": "AIzaSyCsk3J8WFJ5LEMPbhNooarHGiQ-vrrbZGE",
              "locationType": "(cities)",
              "name": "location",
              "onChange": undefined,
              "onDropdownSelect": [Function],
              "placeholder": "San Diego,CA",
              "targetArea": "City, State",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowSameDay": false,
              "dateFormat": "L",
              "dateFormatCalendar": "MMMM YYYY",
              "disabled": false,
              "disabledKeyboardNavigation": false,
              "dropdownMode": "scroll",
              "monthsShown": 1,
              "onBlur": [Function],
              "onChange": [Function],
              "onClickOutside": [Function],
              "onFocus": [Function],
              "onKeyDown": [Function],
              "onMonthChange": [Function],
              "onSelect": [Function],
              "onYearChange": [Function],
              "preventOpenOnFocus": false,
              "selected": "2018-06-06T04:03:09.129Z",
              "shouldCloseOnSelect": true,
              "showTimeSelect": false,
              "timeCaption": "Time",
              "timeIntervals": 30,
              "withPortal": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select Event",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Select Event",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Search Events
</h5>,
          <Async
            autoload={false}
            cache={Object {}}
            ignoreAccents={true}
            ignoreCase={true}
            loadOptions={[Function]}
            loadingPlaceholder="Loading..."
            name="searchEventsInput"
            onChange={[Function]}
            options={Array []}
            placeholder="Search events"
            searchPromptText="Type to search"
            value="San Diego,CA"
>
            [Function]
</Async>,
          <div
            className="eventsSearchLocationDate"
>
            
            <LocationAutocomplete
                        googleAPIKey="AIzaSyCsk3J8WFJ5LEMPbhNooarHGiQ-vrrbZGE"
                        locationType="(cities)"
                        name="location"
                        onChange={undefined}
                        onDropdownSelect={[Function]}
                        placeholder="San Diego,CA"
                        targetArea="City, State"
            />
            <DatePicker
                        allowSameDay={false}
                        dateFormat="L"
                        dateFormatCalendar="MMMM YYYY"
                        disabled={false}
                        disabledKeyboardNavigation={false}
                        dropdownMode="scroll"
                        monthsShown={1}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClickOutside={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onMonthChange={[Function]}
                        onSelect={[Function]}
                        onYearChange={[Function]}
                        preventOpenOnFocus={false}
                        selected={"2018-06-06T04:03:09.129Z"}
                        shouldCloseOnSelect={true}
                        showTimeSelect={false}
                        timeCaption="Time"
                        timeIntervals={30}
                        withPortal={false}
            />
</div>,
          <button
            onClick={[Function]}
>
            Select Event
</button>,
        ],
        "className": "eventsSearchContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Events",
          },
          "ref": null,
          "rendered": "Search Events",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoload": false,
            "cache": Object {},
            "children": [Function],
            "ignoreAccents": true,
            "ignoreCase": true,
            "loadOptions": [Function],
            "loadingPlaceholder": "Loading...",
            "name": "searchEventsInput",
            "onChange": [Function],
            "options": Array [],
            "placeholder": "Search events",
            "searchPromptText": "Type to search",
            "value": "San Diego,CA",
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <LocationAutocomplete
                googleAPIKey="AIzaSyCsk3J8WFJ5LEMPbhNooarHGiQ-vrrbZGE"
                locationType="(cities)"
                name="location"
                onChange={undefined}
                onDropdownSelect={[Function]}
                placeholder="San Diego,CA"
                targetArea="City, State"
/>,
              <DatePicker
                allowSameDay={false}
                dateFormat="L"
                dateFormatCalendar="MMMM YYYY"
                disabled={false}
                disabledKeyboardNavigation={false}
                dropdownMode="scroll"
                monthsShown={1}
                onBlur={[Function]}
                onChange={[Function]}
                onClickOutside={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMonthChange={[Function]}
                onSelect={[Function]}
                onYearChange={[Function]}
                preventOpenOnFocus={false}
                selected={"2018-06-06T04:03:09.129Z"}
                shouldCloseOnSelect={true}
                showTimeSelect={false}
                timeCaption="Time"
                timeIntervals={30}
                withPortal={false}
/>,
            ],
            "className": "eventsSearchLocationDate",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "googleAPIKey": "AIzaSyCsk3J8WFJ5LEMPbhNooarHGiQ-vrrbZGE",
                "locationType": "(cities)",
                "name": "location",
                "onChange": undefined,
                "onDropdownSelect": [Function],
                "placeholder": "San Diego,CA",
                "targetArea": "City, State",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowSameDay": false,
                "dateFormat": "L",
                "dateFormatCalendar": "MMMM YYYY",
                "disabled": false,
                "disabledKeyboardNavigation": false,
                "dropdownMode": "scroll",
                "monthsShown": 1,
                "onBlur": [Function],
                "onChange": [Function],
                "onClickOutside": [Function],
                "onFocus": [Function],
                "onKeyDown": [Function],
                "onMonthChange": [Function],
                "onSelect": [Function],
                "onYearChange": [Function],
                "preventOpenOnFocus": false,
                "selected": "2018-06-06T04:03:09.129Z",
                "shouldCloseOnSelect": true,
                "showTimeSelect": false,
                "timeCaption": "Time",
                "timeIntervals": 30,
                "withPortal": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select Event",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Select Event",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
